dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -413,51 +433,151 @@ def _update_histogram(self, df_series):
+        new_bin_id = 0
+        hist_loss = 0
+        for bin_id, bin_count in enumerate(bin_counts):
+            if not bin_count:  # if nothing in bin, nothing to add
+                continue
+
+            bin_edge = bin_edges[bin_id: bin_id + 3]
+
+            # if we know not float, we can assume values in bins are integers.
+            is_float_profile = self.__class__.__name__ == 'FloatColumn'
+            if not is_float_profile:
+                bin_edge = np.round(bin_edge)
+
+            # loop until we have a new bin which contains the current bin.
+            while (bin_edge[0] >= new_bin_edges[new_bin_id + 1]
+                   and new_bin_id < suggested_bin_count - 1):
+                new_bin_id += 1
+
+            new_bin_edge = new_bin_edges[new_bin_id: new_bin_id + 3]
+            
+            # find where the current bin falls within the new bins
+            is_last_bin = new_bin_id == suggested_bin_count -1
+            if bin_edge[1] < new_bin_edge[1] or is_last_bin:
+                # current bin is within the new bin
+                new_bin_counts[new_bin_id] += bin_count
+                hist_loss += ((
+                    (new_bin_edge[1] + new_bin_edge[0])
+                    - (bin_edge[1] + bin_edge[0])) / 2) ** 2 * bin_count
+            elif bin_edge[0] < new_bin_edge[1]:
+                # current bin straddles two of the new bins
+                # get the percentage of bin that falls to the left
+                percentage_in_left_bin = (
+                    (new_bin_edge[1] - bin_edge[0])
+                    / (bin_edge[1] - bin_edge[0])
+                )
+                count_in_left_bin = round(bin_count * percentage_in_left_bin)
+                new_bin_counts[new_bin_id] += count_in_left_bin
+                hist_loss += ((
+                    (new_bin_edge[1] + new_bin_edge[0])
+                    - (bin_edge[1] + bin_edge[0])) / 2) ** 2 * count_in_left_bin
+
+                # allocate leftovers to the right bin
+                new_bin_counts[new_bin_id + 1] += bin_count - count_in_left_bin
+                hist_loss += ((
+                    (new_bin_edge[2] - new_bin_edge[1])
+                    - (bin_edge[1] - bin_edge[0])
+                ) / 2)**2 * (bin_count - count_in_left_bin)
+
+                # increment bin id to the right bin
+                new_bin_id += 1
+
+        return ({'bin_edges': new_bin_edges, 'bin_counts': new_bin_counts},
+                hist_loss)
+
+    def _get_best_histogram_for_profile(self):
+        """
+        Converts the stored histogram into the presentable state based on the
+        suggested histogram bin count from numpy.histograms. The bin count used
+        is stored in 'suggested_bin_count' for each method.
 
-    def _get_percentile(self, percentile):
+        :return: histogram bin edges and bin counts
+        :rtype: dict
+        """
+        if self._profile_histogram is not None:
+            return self._profile_histogram
```

Comment:
Would this break on update?

Reply:
`update` sets it to `None` such that it has to be recalculated after an `update`